sos/report/plugins/scsi.py

```diff
@@ -39,6 +40,14 @@ def setup(self):
             "/sys/class/scsi_generic"
         ])
 
+        scsi_types = ["enclosu"]
+        result = self.collect_cmd_output('lsscsi -g')
+        if result['status'] == 0:
+            for line in result['output'].splitlines():
+                if (line.split()[1] in scsi_types):
+                    devsg = line.split()[-1]
+                    self.add_cmd_output("sg_ses -p2 -b1 %s" % devsg)
```

Comment:
We can leverage a predicate here to control the command execution based on the presence of the `sg3_utils` package. E.G.

```python
sg3_pred = SoSPredicate(packages='sg3_utils')
[...]
        scsi_types = ["enclosu"]
        result = self.collect_cmd_output('lsscsi -g')
        if result['status'] == 0:
            for line in result['output'].splitlines():
                if (line.split()[1] in scsi_types):
                    devsg = line.split()[-1]
                    self.add_cmd_output("sg_ses -p2 -b1 %s" % devsg,
                                        pred=sg3_pred)
```

Reply:
The other `sg_` commands used in the plugin belong to the same sg3_utils RPM. I think we can live without the predicate in that case.

Reply:
I was actually originally going to suggest adding the predicate to those commands, but then it appears we don't support `pred` as an arg for `add_blockdev_cmd()` currently.

So, I have an AI on me to add that support, and then I'll make a follow up commit that reuses the predicate there as well too.